src/main/kotlin/com/bugsnag/android/gradle/BugsnagUploadSharedObjectTask.kt

```diff
@@ -29,19 +29,24 @@ import javax.inject.Inject
 /**
  * Task that uploads shared object mapping files to Bugsnag.
  */
-internal open class BugsnagUploadNdkTask @Inject constructor(
+internal open class BugsnagUploadSharedObjectTask @Inject constructor(
     objects: ObjectFactory,
     projectLayout: ProjectLayout
 ) : DefaultTask(), AndroidManifestInfoReceiver, BugsnagFileUploadTask {
 
+    enum class UploadType {
+        NDK,
+        UNITY
+    }
```

Comment:
Is it worth putting the endpoint and upload key as methods on this class, so the conditionals aren't needed later on? e.g.

```kotlin
enum class UploadType {
    NDK {
        override fun endpoint(base: String) = "${base}/so-symbol"
        override fun uploadKey() = "soSymbolFile"
    },

    UNITY {
        override fun endpoint(base: String) = "${base}/so-symbol-table"
        override fun uploadKey() = "soSymbolTableFile"
    };

    abstract fun endpoint(base: String): String
    abstract fun uploadKey(): String
}
```

then using it is just

```kotlin
val requestEndpoint = uploadType.get().endpoint(endpoint.get())
val soUploadKey = uploadType.get().uploadKey()
```

Reply:
Or even

```kotlin
enum class UploadType(private val path: String, val uploadKey: String) {
    NDK("so-symbol", "soSymbolFile"),
    UNITY("so-symbol-table", "soSymbolTableFile");

    fun endpoint(base: String): String {
        return "${base}/${path}"
    }
}

val requestEndpoint = uploadType.get().endpoint(endpoint.get())
val soUploadKey = uploadType.get().uploadKey
```

Reply:
good idea 👍  I've altered the enum accordingly